lib/react_client.dart

```diff
@@ -734,8 +746,54 @@ class ReactJsComponentFactoryProxy extends ReactComponentFactoryProxy {
   }
 }
 
+class ReactDartFunctionComponentFactoryProxy extends ReactComponentFactoryProxy
+    with JsBackedMapsComponentFactoryBuilder {
+  /// The ReactJS function used as the type for all [ReactElement]s built by
+  /// this factory.
+  final JsFunctionComponent reactFunction;
+
+  /// The name of this function.
+  final String displayName;
+
+  /// The JS component factory used by this factory to build [ReactElement]s.
+  final ReactJsComponentFactory reactComponentFactory;
+
+  ReactDartFunctionComponentFactoryProxy(this.reactFunction, [this.displayName])
+      : this.reactComponentFactory = React.createFactory(reactFunction);
+
+  get type => reactFunction;
+}
+
+/// Creates a function component from the given [dartFunctionComponent] that can be used with React.
+JsFunctionComponent _wrapFunctionComponent(DartFunctionComponent dartFunctionComponent, {String displayName}) {
+  assert(() {
+    displayName ??= getProperty(dartFunctionComponent, 'name');
+    return true;
+  }());
+  jsFunctionComponent(JsMap jsProps, [JsMap _legacyContext]) =>
+      (dartFunctionComponent(jsProps != null ? JsBackedMap.backedBy(jsProps) : JsBackedMap()) ?? jsNull);
+  JsFunctionComponent interopFunction = allowInterop(jsFunctionComponent);
+  if (displayName != null) {
+    // This is a work-around to display the correct name in the React DevTools.
+    callMethod(getProperty(window, 'Object'), 'defineProperty', [
```

Comment:
Can we use a JS-interop binding for this? Those read better and have better static analysis.
```dart
@JS('Object.defineProperty')
external void _defineProperty(dynamic object, String propertyName, dynamic descriptor);
```

Reply:
![c75](https://user-images.githubusercontent.com/41018730/67874774-a9d74d80-faf2-11e9-8688-8908cdcc2448.gif)